iOS Meetings (#265)
Babylonpartners/ios-playbook: Cookbook/Technical-Documents/meetings-purpose.md

# iOS Chapter Meetings

> As an engineer part of the iOS Chapter Team we have to collaborate with each other. The work we do in each of our squads contributes to the whole product and at the end of the Sprint we are all releasing the same app, together, as a team. 

Reviewer: ```suggestion
As an engineer part of the iOS Chapter Team we have to collaborate with each other. The work we do in our squads contributes to the whole product and at the end of the Sprint we are all releasing the same app, together as a team. 
```

> We should all be focused in delivering the work assigned from our squad and at the same time we should try to engage with the other iOS Engineers to keep the communication channel open and make everyone feel comfortable to raise any comment/question.

Reviewer: ```suggestion
We should all be focused in delivering the work assigned from our squad and at the same time try to engage with the other iOS Engineers to keep the communication channel open and make everyone feel comfortable to raise any comment/question.
```

We have scheduled a set of iOS Team specific meetings to ensure we communicate with each other, raise any issue, announce important events and are aware of the process and follow it.

All the outcomes from these meetings are posted in the #ios-meeting-outcomes Slack channel along with their recording.

## iOS Retro

> The place where we can raise anything that went well or should be improved in our last 2 weeks of work. This can be anything related with the iOS Team in general or something from our Squads in particular but that we feel is relevant to raise. 

Reviewer: ```suggestion
The place where we can raise anything that went well or should be improved in our last 2 weeks of work. This can be anything related to the iOS Team in general or something from our Squads in particular but we feel relevant to raise. 
```

Some of the issues you have in your squad are common to other squads. Those can be raised in this meeting to understand if it’s something that can be addressed. Additionally, the good achievements we have in our squads can and should also be raised in the retro so everyone is aware of the good work everyone is doing.

This is a fortnightly meeting with 30 minutes duration.
The schedule is the following:

> * For 2 weeks the board will be available in the [Trello Engineering Team](https://trello.com/iosengineeringteam) (Please look for the board with the title "Retro \<Date of the Retro>") so we can add items at any time during those 2 weeks.

Reviewer: ```suggestion
* For 2 weeks the board will be available in the [Trello iOS Engineering Team](https://trello.com/iosengineeringteam) (Please look for the board with the title "Retro \<Date of the Retro>") so we can add items at any time during those 2 weeks.
```

* The first 2 minutes will be for us to vote. (Tip: Hovering over a card and clicking "v" will add or remove a vote.)
* The next 28 minutes will be to go through the most voted items in the **Went Well** and **To Improve** columns and add any required action to the **Action Items** column. Each action has to have someone assigned to it.

## PR Parties

> The PR Parties are meetings where a group of engineers will get together to review a bunch of PRs pre-defined. Usually the authors of these PRs will be in the group and they will go through their PRs and explain them.

Reviewer: ```suggestion
The PR Parties are meetings where a group of engineers will get together to review a bunch of PRs they're assigned to. Usually the authors of these PRs will be in the group and they will go through their PRs and explain them.
```

This will allow the authors to explain the solution in the PR, to open the discussion about it and to agree in an approach if required. In the ideal world, we would have all the PRs approved in the end of the meeting, but this is not the main purpose of the meeting. The purpose is to facilitate the discussion about the PR implementation.

A PR Party Line-Up will be announced every Monday, Wednesday and Thursday.

* The Line-Up will be posted in the #ios-meeting-outcomes Slack channel.
* Each PR Party will be composed by 3-4 engineers and will have 3-4 PRs to review.
* The PR Party schedule and duration will be decided by the group.
* One of the engineers in each group will be the organizer (nominated by who defines the group). The organizer's responsability is to:

>     - create the slack group channel with the other participants

Reviewer: ```suggestion
    - create the Slack channel with the other participants
```

    - make sure a schedule is defined
    - kick-off the meeting
    - post the outcome in the thread of the PR Party Line-Up message in Slack.
* There is an exception in the final week of each sprint. The Main PR Party on Thursday will still happen to accelerate PR Reviews for release. In these weeks there will only be 2 sets of 2 groups of PR Parties.

## iOS Lunch & Learn

The purpose of this meeting is to have fortnightly iOS learning sessions with the whole team.
We will grab lunch before the meeting, sit together (physically or remotelly) and:

* Watch a video relevant to the iOS team.
* Watch a presentation done by someone in the team (full or lightning).
* Have a discussion about a topic that impacts the iOS team.

> Every meeting has an organizer that is responsible for:

Reviewer: ```suggestion
Every meeting has an organizer who is responsible for:
```

* Setting up the zoom call
* Book a room if in the office
* Define a topic.

The schedule can be found [here](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit#gid=0).

Topics to be picked up can be found [here](https://docs.google.com/spreadsheets/d/1kdY3edy_TeqIGH_7VnZzElxgVo_qD2z4EF-arWNShyw/edit#gid=1037493294).

## Banter

> An unstrutured meeting happening every week for 30 minutes. It is the place for us to speak about any topic we would like, can even be something completely unrelated with work. 

Reviewer: ```suggestion
An unstructured meeting happening every week for 30 minutes. It is the place for us to speak about any topic we would like, can even be something completely unrelated with work.
```

> It is frequently used to make small announcements and to have small discussions around a recent event.

Reviewer: ```suggestion
It is frequently used to make small announcements and have small discussions around a recent event.
```

# iOS Chapter Meetings

As an engineer part of the iOS Chapter Team we have to collaborate with each other. The work we do in each of our squads contributes to the whole product and at the end of the Sprint we are all releasing the same app, together, as a team. 

> We should all be focused in delivering the work assigned from our squad and at the same time we should try to engage with the other iOS Engineers to keep the communication channel opened and to make everyone feel comfortable to raise any comment/question.

Reviewer 2: ```suggestion
We should all be focused in delivering the work assigned from our squad and at the same time we should try to engage with the other iOS Engineers to keep the communication channel open and make everyone feel comfortable to raise any comment/question.
```

We have scheduled a set of iOS Team specific meetings to ensure we communicate with each other, raise any issue, announce important events and are aware of the process and follow it.

> All the outcomes from these meetings are posted in the #ios-meeting-outcomes Slack channel along with it's recording.

Reviewer: ```suggestion
All the outcomes from these meetings are posted in the #ios-meeting-outcomes Slack channel along with its recording.
```

> All the outcomes from these meetings are posted in the #ios-meeting-outcomes Slack channel along with its recording.

Reviewer 2: ```suggestion
All the outcomes from these meetings are posted in the #ios-meeting-outcomes Slack channel along with their recording.
```

## iOS Retro

The place where we can raise anything that went well or should be improved in our last 2 weeks of work. This can be anything related with the iOS Team in general or something from our Squads in particular but that we feel is relevant to raise. 

Some of the issues you have in your squad are common to other squads. Those can be raised in this meeting to understand if it’s something that can be addressed. Additionally, the good achievements we have in our squads can and should also be raised in the retro so everyone is aware of the good work everyone is doing.

This is a fortnightly meeting with 30 minutes duration.
The schedule is the following:

* For 2 weeks the board will be available in the [Trello Engineering Team](https://trello.com/iosengineeringteam) (Please look for the board with the title "Retro \<Date of the Retro>") so we can add items at any time during those 2 weeks.
* The first 2 minutes will be for us to vote. (Tip: Hovering over a card and clicking "v" will add or remove a vote.)

> * The next 28 minutes will be to go through the most voted items in the **Went Well** and **To Improve** columns and add any required action to the **Action Items** column. Each action have to have someone assigned to it.

Reviewer 2: ```suggestion
* The next 28 minutes will be to go through the most voted items in the **Went Well** and **To Improve** columns and add any required action to the **Action Items** column. Each action has to have someone assigned to it.
```

## PR Parties

The PR Parties are meetings where a group of engineers will get together to review a bunch of PRs pre-defined. Usually the authors of these PRs will be in the group and they will go through their PRs and explain them.
This will allow the authors to explain the solution in the PR, to open the discussion about it and to agree in an approach if required. In the ideal world, we would have all the PRs approved in the end of the meeting, but this is not the main purpose of the meeting. The purpose is to facilitate the discussion about the PR implementation.

A PR Party Line-Up will be announced every Monday, Wednesday and Thursday.

* The Line-Up will be posted in the #ios-meeting-outcomes Slack channel.
* Each PR Party will be composed by 3-4 engineers and will have 3-4 PRs to review.
* The PR Party schedule and duration will be decided by the group.

> * One of the engineers in each group will be the organizer (nominated by who defines the group). The organizer responsability is to :

Reviewer: ```suggestion 
* One of the engineers in each group will be the organizer (nominated by who defines the group). The organizer’s responsibility is to:
```

    - create the slack group channel with the other participants
    - make sure a schedule is defined
    - kick-off the meeting
    - post the outcome in the thread of the PR Party Line-Up message in Slack.
* There is an exception in the final week of each sprint. The Main PR Party on Thursday will still happen to accelerate PR Reviews for release. In these weeks there will only be 2 sets of 2 groups of PR Parties.

## iOS Lunch & Learn

> The purpose of this meeting is to have fortnight iOS learn sessions with the whole team.

Reviewer: ```suggestion
The purpose of this meeting is to have fortnightly iOS learning sessions with the whole team.
```

> We will grab lunch before the meeting, we will sit together (physically or remotelly) and we will: 

Reviewer: ```suggestion
We will grab lunch before the meeting and sit together (physically or remotely) to: 
```
